staff/jordi-pujol/e-commerce-protos/scripts/cart.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Abstraction of Cart
+ * 
+ * @method (add)
+ *  @param {Object} object
+ *  @throws {TypeError}
+ *  @returns {Number} Array length
+ * 
+ */
+
+function Cart() {
+    this.cartItems = [];
+}
+    
+    Cart.prototype.add = function (item){
+
+        if (!(item instanceof Product)) throw TypeError ('Should be an object');
+
+        this.cartItems.push(item)
+        return this.cartItems.length
+    };
+
+    Cart.prototype.totalPrice = function(){
+        var sumP = 0;
+        for (let i = 0; i< this.cartItems.length; i++){
+           sumP += this.cartItems[i].price
+        };
+        return sumP
+    };
+
+    Cart.prototype.numberOfItems = function (){
+        sumItems = 0;
+        for (let i =0; i< this.cartItems.length; i++){
+            sumItems ++
+        }
+        return sumItems
+    }
+
+    Cart.prototype.mostExpensive = function() {
+        var exp = this.cartItems[0].price;
+        var pro = {}
+        for (let i=0; i< this.cartItems.length; i++){
+            if (exp < this.cartItems[i].price) {
+                exp = this.cartItems[i].price
+                pro = this.cartItems[i].constructor.name
```

Comment:
Are you sure that with the constructor.name you get the name of the product?

Reply:
Well, you are right I get the name of the constructor, but not the product one...